internal/client.go

```diff
@@ -352,6 +352,25 @@ type (
 		// GetWorkerTaskReachability returns which versions are is still in use by open or closed workflows.
 		GetWorkerTaskReachability(ctx context.Context, options *GetWorkerTaskReachabilityOptions) (*WorkerTaskReachability, error)
 
+		// DescribeTaskQueueEnhanced returns information about the target task queue, broken down by Build Id:
+		//   - List of pollers
+		//   - Workflow Reachability status
+		//   - Backlog info for Workflow and/or Activity tasks
+		// When not supported by the server, it returns an empty [TaskQueueDescription]
```

Comment:
Doesn't the implementation return an error?

Reply:
The server does not return an error, at the wire level we use the old `describe` api and all the `enhanced fields` in the protobuf are just ignored. The result returned could be non-empty but it is likely incorrect because some of the ignored fields change the query. What we do is just ignore these values when deserializing the protobuf and return an empty TaskQueueDescription.

Reply:
Sorry maybe I am missing something,  but doesn't the [implementation ](https://github.com/temporalio/sdk-go/pull/1494/files#diff-e86803d9c6ede32f27826e8eff45f92375372cb213b6b46e1c0513633e2957ccR1062)of `DescribeTaskQueueEnhanced` call  `detectTaskQueueEnhancedNotSupported` and that will  return an error if it is not supported?

Reply:
I need to be more explicit in the docs. `detectTaskQueueEnhancedNotSupported`  is a best effort function. If there are no pollers and no taskqueue status for the requested queue the response is empty, so I cannot tell... Otherwise, yes, I get an error.

Reply:
Yeah may be better to be consistent then if we can't always return an error if it is not supported

Reply:
Thinking about this a bit more. The only case that we cannot detect is the empty return case, but we get correct responses regardless if we return empty.
 All the other cases are handled ok by `detectTaskQueueEnhancedNotSupported` and we get an error, which gives as more useful info than just returning empty.
I'm inclined to keep it "inconsistent" but make it more clear in the docs. In fact, it is more "incomplete" than "inconsistent" because we always get correct answers...

Reply:
something like this:
// When not supported by the server, it returns an empty [TaskQueueDescription] if there is no information
// about the task queue, or an error when the response identifies an unsupported server.

Reply:
works for me